### ISSUE TITLE
ESLint - 'window' is already defined as a built-in global variable

### ISSUE THREAD

**Issue by JurajKavka** [team-innovation/vue-sfc-rollup#32]
When I enable eslint recommended rules, linting ends with error in `entry.js` file

```
/xxxxxxxx/src/entry.js
  17:11  error  'window' is already defined as a built-in global variable  no-redeclare

✖ 1 problem (1 error, 0 warnings)
``` 

My `.eslintrc`:

```
module.exports = {
  extends: [
    // add more generic rulesets here, such as:
    'eslint:recommended',
    'plugin:vue/strongly-recommended'
  ],
  rules: {
    // override/add rules settings here, such as:
    // 'vue/no-unused-vars': 'error'
  }
}
```

I'm not sure if it is a bug so I have made exception to ignore this error before line no.17 in `entry.js`

```
// eslint-disable-next-line no-redeclare
/* global window global */
let GlobalVue = null;
```

**Comment by mgdodge** [comment]
Looks like that line was fine in eslint 5.x, but the eslint 6.x update is more strict with some rules, and this is one example. Your solution is their recommended one as well. 👍 

You bring up a good point, the template output should probably be able to pass the default eslint rules for the vue plugin, so I will get a patch-level update ready that fixes linting. But what you have here is probably what I will end up having in place for this particular error.

**Comment by JurajKavka** [comment]
Thank You for Your response :) 

I just want to ask here.. is there some recommended way to set up test enviroment in `vue-sfc-rollup` project? With Jest or Mocha. I found this predefined in `vue-cli` projects, but I don't know how to make Vue library there. With `vue-sfc-rollup` is making Vue component library super easy but i don't know how to setup tests. Thank You! ;)

**Comment by mgdodge** [comment]
Using the vue cli alongside vue-sfc-rollup is fine. In fact, as of v2.2.0, there is a @vue/cli-service dependency so it is expected that is how rapid development will happen.

The easiest way to set up testing is to do the following:

1. `npm install --save-dev @vue/cli-plugin-unit-jest @vue/test-utils`
2. Add `"test": "vue-cli-service test:unit"` to the `scripts` section of `package.json`
3. Create test files following the naming convention `root-directory/tests/unit/component.spec.ts`

With those pieces in place, the vue cli will intelligently set up everything else it needs, detect the test files, and you will be up and running.

**Comment by JurajKavka** [comment]
Thank You!! With Your kick and with help of [this](https://vue-test-utils.vuejs.org/guides/testing-single-file-components-with-jest.html), my tests with Jest are working! :)

**Comment by mgdodge** [comment]
Closed with v2.2.2
